libbeat/cmd/instance/metrics.go

```diff
@@ -55,13 +59,29 @@ func setupMetrics(name string) error {
 		IncludeTop:   process.IncludeTopConfig{},
 	}
 	err := beatProcessStats.Init()
+	if err != nil {
+		return err
+	}
+
+	beatProcessSysInfo, err = sysinfo.Self()
+	if err != nil {
+		return err
+	}
 
-	return err
+	var ok bool
+	handleCounter, ok = beatProcessSysInfo.(types.OpenHandleCounter)
+	if !ok {
+		return fmt.Errorf("cannot convert process to OpenHandleCounter: %v", beatProcessSysInfo)
+	}
+
+	return nil
 }
 
 func setupPlatformSpecificMetrics() {
 	if runtime.GOOS != "windows" {
 		monitoring.NewFunc(systemMetrics, "load", reportSystemLoadAverage, monitoring.Report)
+	} else {
+		monitoring.NewFunc(beatMetrics, "file_handles", reportOpenHandles, monitoring.Report)
```

Comment:
I think we should be moving towards calling this "handles" rather than "file handles" since it includes sockets / pipes.